test/framework/src/main/java/org/elasticsearch/test/disruption/DisruptableMockTransport.java

```diff
@@ -42,58 +59,79 @@ public DisruptableMockTransport(Logger logger) {
         this.logger = logger;
     }
 
-    protected abstract DiscoveryNode getLocalNode();
+    public abstract DiscoveryNode getLocalNode();
 
-    protected abstract ConnectionStatus getConnectionStatus(DiscoveryNode sender, DiscoveryNode destination);
+    protected abstract ConnectionStatus getConnectionStatus(DiscoveryNode destination);
 
-    protected abstract Optional<DisruptableMockTransport> getDisruptedCapturingTransport(DiscoveryNode node, String action);
+    protected abstract Optional<DisruptableMockTransport> getDisruptableMockTransport(TransportAddress address);
 
-    protected abstract void handle(DiscoveryNode sender, DiscoveryNode destination, String action, Runnable doDelivery);
+    protected abstract void schedule(Runnable runnable);
```

Comment:
I think the name `execute` would be more consistent with things like `ExecutorService`. `schedule` is largely used for delayed execution (ignoring that the `DeterministicTaskQueue` uses `scheduleNow` for this).